receiver/opentelemetryreceiver/metrics/opentelemetry.go

```diff
@@ -0,0 +1,124 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/observability"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+	opentelemetrytranslator "github.com/open-telemetry/opentelemetry-collector/translator/metrics/opentelemetry"
+	collectormetricspb "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/metrics/v1"
+	metricspb "github.com/open-telemetry/opentelemetry-proto/gen/go/metrics/v1"
+	"go.opencensus.io/trace"
+	"google.golang.org/api/support/bundler"
+)
+
+// Receiver is the type used to handle metrics from OpenCensus exporters.
+type Receiver struct {
+	nextConsumer       consumer.MetricsConsumer
+	metricBufferPeriod time.Duration
+	metricBufferCount  int
+}
+
+// New creates a new ocmetrics.Receiver reference.
+func New(nextConsumer consumer.MetricsConsumer, opts ...Option) (*Receiver, error) {
+	if nextConsumer == nil {
+		return nil, oterr.ErrNilNextConsumer
+	}
+	r := &Receiver{nextConsumer: nextConsumer}
+	for _, opt := range opts {
+		opt.WithReceiver(r)
+	}
+	return r, nil
+}
+
+var _ collectormetricspb.MetricsServiceServer = (*Receiver)(nil)
+
+const receiverTagValue = "otlp_metrics"
+
+func (r* Receiver) Export(ctx context.Context, req *collectormetricspb.ExportMetricsServiceRequest) (*collectormetricspb.ExportMetricsServiceResponse, error) {
+	// The bundler will receive batches of metrics i.e. []*metricspb.Metric
+	// We need to ensure that it propagates the receiver name as a tag
+	ctxWithReceiverName := observability.ContextWithReceiverName(ctx, receiverTagValue)
+	metricsBundler := bundler.NewBundler((*consumerdata.MetricsData)(nil), func(payload interface{}) {
+		r.batchMetricExporting(ctxWithReceiverName, payload)
+	})
+
+	metricBufferPeriod := r.metricBufferPeriod
+	if metricBufferPeriod <= 0 {
+		metricBufferPeriod = 2 * time.Second // Arbitrary value
+	}
+	metricBufferCount := r.metricBufferCount
+	if metricBufferCount <= 0 {
+		// TODO: (@odeke-em) provide an option to disable any buffering
+		metricBufferCount = 50 // Arbitrary value
+	}
+
+	metricsBundler.DelayThreshold = metricBufferPeriod
+	metricsBundler.BundleCountThreshold = metricBufferCount
+
+	for _, resourceMetrics := range req.ResourceMetrics {
+		err := processReceivedMetrics(resourceMetrics, metricsBundler)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &collectormetricspb.ExportMetricsServiceResponse{}, nil
+}
+
+func processReceivedMetrics(resourceMetrics *metricspb.ResourceMetrics, bundler *bundler.Bundler) error {
+	if len(resourceMetrics.Metrics) == 0 {
+		return nil
+	}
+
+	md, err := opentelemetrytranslator.ResourceMetricsToMetricsData(resourceMetrics)
+	if err != nil {
+		return nil
+	}
+	return bundler.Add(md, len(md.Metrics))
+}
+
+func (r *Receiver) batchMetricExporting(longLivedRPCCtx context.Context, payload interface{}) {
```

Comment:
TODO: fix argument name, `longLivedRPCCtx` was probably referring to streaming API